docs/design/otlp.md

```diff
@@ -0,0 +1,171 @@
+| `gzip-tonic` `zstd-tonic` | gRPC message compression | `exporter/tonic` |
+| `reqwest-client` / `reqwest-blocking-client` *(default)* | Choose async vs blocking Reqwest HTTP client |
+| `hyper-client` | Use Hyper HTTP transport *(requires both Reqwest features disabled)* |
+| TLS helpers (`tls-roots`, `tls-webpki-roots`, `reqwest-rustls`) | Supply trust roots for TLS |
+
+Because **only one transport is valid per exporter**, `protocol` is currently a *hint* – unsupported values are ignored by the concrete builder.
+
+### 4.1 Transport-specific Client Options
+
+Exporters must function both in binaries that start no async runtime and in services already running Tokio.  Consequently the crate provides **blocking** and **non-blocking** HTTP clients behind mutually-exclusive feature-flags.  The blocking variant is the default to ensure out-of-the-box operability; applications that already depend on Tokio can opt into an async client to avoid spawning extra threads.
+
+| Implementation | Feature flag | Blocking? | Underlying crate | Transports | Typical use case |
+|---------------|--------------|-----------|------------------|-----------|------------------|
+| Reqwest (blocking) **(default)** | `reqwest-blocking-client` | Yes | `reqwest::blocking` | HTTP/1.1 → auto-upgrades to HTTP/2 over TLS | CLI tools, synchronous binaries, or hybrid apps where introducing Tokio is undesirable. A helper thread is spawned once at builder time.
+| Reqwest (async) | `reqwest-client` | No (Tokio) | `reqwest` | HTTP/1.1 → auto-upgrades to HTTP/2 over TLS | Services that already run a Tokio runtime and prefer fully non-blocking exports.
+| Hyper | `hyper-client` | No (Tokio) | `hyper` | HTTP/1.1 → auto-upgrades to HTTP/2 over TLS | Lean dependency footprint when both Reqwest features are disabled.
+| Tonic (gRPC) | `grpc-tonic` | No (Tokio) | `tonic` | HTTP/2 (gRPC) | Environments standardising on gRPC or collectors exposing only port 4317.
+| Custom | — | Depends | user-supplied | depends on implementation | Embedding a bespoke `HttpClient` via `.with_http_client()`.
+
+---
+
+## 5. Runtime Flow (Trace Export)
+```mermaid
+sequenceDiagram
+    participant App as Application code
+    participant SDK as opentelemetry-sdk
+    participant Exp as SpanExporter
+    participant Tx as Transport Client (HTTP/Reqwest or gRPC/Tonic)
+    participant Collector as OTLP Collector
+    App->>SDK: span.start / span.end
+    SDK-->>SDK: BatchSpanProcessor collects SpanData
+    SDK->>Exp: export(batch)
+    Exp->>Tx: serialize + send
+    Tx--)Collector: OTLP request
+    note right of Collector: 2xx = success,<br/>non-2xx → error handling
+    Tx--)Exp: Future resolves
+    Exp--)SDK: OTelSdkResult
+```
+Notes: 
+
+* Serialization happens **inside the transport client** module to keep exporters thin.
+* `export` is `async` for `tonic` / `reqwest` clients, surfaced via `SpanExporter` implementing `opentelemetry_sdk::trace::SpanExporter`.
```

Comment:
is it correct to state exporter is "async" for only tonic/reqwest? the trait method for export returns a Future...